google/cloud/forseti/scanner/audit/enabled_apis_rules_engine.py

```diff
@@ -0,0 +1,401 @@
+# Copyright 2018 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Enabled APIS rules engine for organizations, folders, and projects.
+
+Builds the RuleBook (EnabledApisRuleBook) from the rule definitions (file either
+stored locally or in GCS) and compares a list of APIs  against the RuleBook to
+determine whether there are violations.
+"""
+
+import collections
+import itertools
+import threading
+
+from google.cloud.forseti.common.gcp_type import errors as resource_errors
+from google.cloud.forseti.common.gcp_type import resource as resource_mod
+from google.cloud.forseti.common.gcp_type import resource_util
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.common.util import relationship
+from google.cloud.forseti.scanner.audit import base_rules_engine as bre
+from google.cloud.forseti.scanner.audit import errors as audit_errors
+
+LOGGER = logger.get_logger(__name__)
+
+# Rule Modes.
+_WHITELIST = 'whitelist'
+_BLACKLIST = 'blacklist'
+_REQUIRED = 'required'
+_RULE_MODES = frozenset([_WHITELIST, _BLACKLIST, _REQUIRED])
+
+# Rule violation types.
+_VIOLATION_TYPES = {
+    _WHITELIST: 'ENABLED_APIS_ADDED_VIOLATION',
```

Comment:
We actually have an open issue to clean this up in the IAM scanner, to make the violation type here easier to understand, i.e. not have `ADDED` or `REMOVED`.  So, can you please change this here to just use `ENABLED_APIS_VIOLATION`.  Then the user can look at the actual rule name in the violation to know what this is for.

Reply:
Fixed.